tests/Microsoft.Identity.Test.Common/Core/Helpers/CoreAssert.cs

```diff
@@ -46,6 +47,31 @@ public static void AreEqual(DateTimeOffset expected, DateTimeOffset actual, Time
             Assert.IsTrue(DictionariesAreEqual(dict1, dict2, valueComparer));
         }
 
+        public static void IsImmutable<T>()
+        {
+            Assert.IsTrue(IsImmutable(typeof(T)));
+        }
+
+        private static bool IsImmutable(Type type)
```

Comment:
>IsImmutable [](start = 28, length = 11)

do we need to make these checks?

Reply:
I like to keep my objects immutable explicitly, it's very easy to add a setter on a property by mistake. The root cause of this bug was mutating the authority

Reply:
ok